clang-tools-extra/clang-doc/Representation.h

```diff
@@ -209,6 +209,9 @@ struct TemplateSpecializationInfo {
 
   // Template parameters applying to the specialized record/function.
   std::vector<TemplateParamInfo> Params;
+
+  // Used to distinguish class specialization file names.
+  std::optional<SmallString<16>> MangledName;
```

Comment:
can you just always use a mangled name for everything? for C mangling would just be the name, but for anything in C++, it'd get you a unique name. Seems reasonable to do unconditionally. WDYT?